src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,48 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+    private F first;
+    private S second;
 
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    static <F, S> Pair<F, S> of(F first, S second) {
+        return new Pair<>(first, second);
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 31;
+        result = 17 * result + ((first == null) ? 0 : first.hashCode());
+        result = 17 * result + ((second == null) ? 0 : second.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Pair.class)) {
+            Pair pair = (Pair) obj;
```

Comment:
we should cast to our parametrize class(ignore idea warnings)

Reply:
Done